modules/products/tomcat/src/main/kotlin/com/metalbear/mirrord/products/tomcat/TomcatExecutionListener.kt

```diff
@@ -22,15 +31,67 @@ private fun getTomcatServerPort(): String {
+ * protected.
+ * @property params first element is script, then args, e.g.:
+ * ["/var/folders/4l/810mmn597cx7zy5w2bk11clh0000gn/T/mirrord-bin/opt/homebrew/Cellar/tomcat@8/8.5.95/libexec/bin/catalina.sh", "run"]
+ */
+class PatchedScriptInfo(
+    scriptsHelper: ScriptHelper?,
+    parent: CommonStrategy,
+    startupNotShutdown: Boolean,
+    private val params: Array<String>
+) :
+    ScriptInfo(scriptsHelper, parent, startupNotShutdown) {
+
+    /** Get the original script object, forcefully change the accessibility of the private parameters field, then write
+     * the patched parameters to that field, and return the changed script object.
+     */
+    override fun getScript(): ExecutableObject? {
+        MirrordLogger.logger.debug("getScript override")
+        val startupScript = super.getScript()
+        MirrordLogger.logger.debug("setting params to: ${params.joinToString(" ")}")
+        val cmdExecutableObject = startupScript as? CommandLineExecutableObject
+        val cmd = cmdExecutableObject ?: return startupScript
+        val myParamsField = CommandLineExecutableObject::class.java.getDeclaredField("myParameters")
+        myParamsField.isAccessible = true
+        myParamsField.set(cmd, params)
+        MirrordLogger.logger.debug("getScript survived forced access")
+        return startupScript
+    }
+}
+
+/**
+ * Create script object that will return a script object of SIP-patched script instead of the original.
+ */
+fun patchedScriptFromScript(scriptInfo: ScriptInfo, script: String, args: String?): PatchedScriptInfo {
+    MirrordLogger.logger.debug("patchedScriptFromScript - script: $script, args: $args")
+    val argList = args?.split("(?<!\\\\) ".toRegex()) ?: emptyList()
+    val params = listOf(script) + argList
+    val parentField = ScriptInfo::class.java.getDeclaredField("myParent")
+    val startupNotShutdownField = ScriptInfo::class.java.getDeclaredField("myStartupNotShutdown")
+    parentField.isAccessible = true
+    startupNotShutdownField.isAccessible = true
+    val parent = parentField.get(scriptInfo) as CommonStrategy
+    val startupNotShutdown = startupNotShutdownField.get(scriptInfo) as Boolean
+    MirrordLogger.logger.debug("patchedScriptFromScript survived forced accesses")
+    return PatchedScriptInfo(scriptInfo.scriptHelper, parent, startupNotShutdown, params.toTypedArray())
+}
+
+data class SavedConfigData(val envVars: List<EnvironmentVariable>, val nonDefaultScript: String?)
+
+data class CommandLineWithArgs(val command: String, val args: String?)
+
 class TomcatExecutionListener : ExecutionListener {
-    private val savedEnvs: ConcurrentHashMap<String, List<EnvironmentVariable>> = ConcurrentHashMap()
+    private val savedEnvs: ConcurrentHashMap<String, SavedConfigData> = ConcurrentHashMap()
 
     private fun getConfig(env: ExecutionEnvironment): RunnerSpecificLocalConfigurationBit? {
-        if (!env.toString().startsWith("Tomcat")) {
-            return null
-        }
```

Comment:
This bit made it so that if you give the configuration a name that does not start with "Tomcat" the whole tomcat code would not be called, so I removed it. If there is some known problem this was supposed to prevent please let me know so that I can try to find another way to solve that problem.